packages/react-components/babel-preset-global-context/src/transformPlugin.ts

```diff
@@ -0,0 +1,259 @@
+import type { NodePath, PluginObj, PluginPass, types as t } from '@babel/core';
+import { types } from '@babel/core';
+import { declare } from '@babel/helper-plugin-utils';
+import hash from '@emotion/hash';
+import * as findUp from 'find-up';
+import { dirname, relative } from 'path';
+import { readFileSync } from 'fs';
+import {
+  CONTEXT_SELECTOR_PACKAGE,
+  CREATE_CONTEXT_CALL,
+  GLOBAL_CONTEXT_ALIAS,
+  GLOBAL_CONTEXT_PACKAGE,
+  GLOBAL_CONTEXT_SELECTOR_ALIAS,
+  GLOBAL_CONTEXT_SELECTOR_PACKAGE,
+  REACT_PACKAGE,
+} from './constants';
+
+type BabelPluginState = PluginPass & {
+  importDeclarationPaths?: NodePath<t.ImportDeclaration>[];
+  nativeExpressionPaths?: NodePath<t.CallExpression>[];
+  contextSelectorExpressionPaths?: NodePath<t.CallExpression>[];
+  nativeLocalName?: string;
+  contextSelectorLocalName?: string;
+};
+
+interface PackageJSON {
+  name: string;
+  version: string;
+}
+
+/**
+ * Checks that passed callee imports react context or context selector
+ */
+function isCreateContextCallee(path: NodePath<t.Expression | t.V8IntrinsicIdentifier>): path is NodePath<t.Identifier> {
+  if (!path.isIdentifier) {
+    return false;
+  }
+
+  return (
+    path.referencesImport(REACT_PACKAGE, CREATE_CONTEXT_CALL) ||
+    path.referencesImport(CONTEXT_SELECTOR_PACKAGE, CREATE_CONTEXT_CALL)
+  );
+}
+
+/**
+ *
+ * @param packageName - name of the package to import
+ * @param functionName - name of the createContext function
+ * @returns a new import expression for global context
+ *
+ * @example import { createContext as __createGlobalContext } from '@global-context/react'
+ */
+function createGlobalContextImportDeclaration(packageName: string, isContextSelector: boolean) {
+  const origFunction = isContextSelector ? 'createContextSelector' : 'createContext';
+  const newFunction = isContextSelector ? GLOBAL_CONTEXT_SELECTOR_ALIAS : GLOBAL_CONTEXT_ALIAS;
+  return types.importDeclaration(
+    [types.importSpecifier(types.identifier(newFunction), types.identifier(origFunction))],
+    types.stringLiteral(packageName),
+  );
+}
+
+/**
+ * @returns - An expression that creates a global context
+ *
+ * @example const MyContext = __createGlobalContext();
+ */
+function createGlobalContextCallExpression(options: {
+  expressionPath: NodePath<t.CallExpression>;
+  packageJson: PackageJSON;
+  packageJsonPath: string;
+  filePath: string;
+  functionName: string;
+}) {
+  const { expressionPath, packageJson, packageJsonPath, filePath, functionName } = options;
+
+  const args = expressionPath.get('arguments').map(arg => arg.node);
+  if (!expressionPath.parentPath.isVariableDeclarator()) {
+    return expressionPath.node;
+  }
+
+  // Use the relative path from package.json because the same package
+  // can be installed under different paths in node_modules if they are duplicated
+  const relativePath = relative(packageJsonPath, filePath);
+  const id = expressionPath.parentPath.get('id') as NodePath<babel.types.Identifier>;
```

Comment:
lets not use global types when we can avoid those

```suggestion
  const id = expressionPath.parentPath.get('id') as NodePath<types.Identifier>;
```